drivers/display/grove_lcd_rgb.c

```diff
@@ -267,16 +267,6 @@ int glcd_initialize(struct device *port)
 		return -EPERM;
 	}
 
-	/* Since device_get_binding() will not return any
-	 * reference to a driver instance if port->driver_api
-	 * is NULL and grove_lcd does not have any API struct,
-	 * just populate it with some magic number
-	 * so grove_lcd can be referenced.
-	 *
-	 * Since dev is probably still in registers.
-	 * use that to minimize code addition.
-	 */
-	port->driver_api = (void *)dev;
```

Comment:
This driver was reusing its driver_data struct as the driver_api?   Is that kosher?  I know that this isn't related to this patch, but this seems like a bug to me.  Both are opaque handles, but the API struct is specific to the driver and driver_data should be one per device.

Reply:
I also had a look at this and actually this driver has no API struct but you can not pass NULL else [device_get_binding](https://github.com/zephyrproject-rtos/zephyr/blob/master/kernel/device.c#L68) will not return the device even if it is available.

Reply:
@andyross : yeah since this driver necessitated setting of driver_api so that device_get_binding() passes, so driver owner just populated it with some magic number so grove_lcd can be referenced. Previously they were using (void *)dev, now i have used (void *)&grove_lcd_driver (struct glcd_data), which is available while using DEVICE_AND_API_INIT()